test/test_common/simulated_time_system.h

```diff
@@ -64,10 +65,27 @@ class SimulatedTimeSystemHelper : public TestTimeSystem {
   class SimulatedScheduler;
   class Alarm;
   friend class Alarm; // Needed to reference mutex for thread annotations.
-  struct CompareAlarms {
-    bool operator()(const Alarm* a, const Alarm* b) const;
+  struct AlarmRegistration {
+    AlarmRegistration(MonotonicTime time, uint64_t randomness, Alarm* alarm)
+        : time_(time), randomness_(randomness), alarm_(alarm) {}
+
+    MonotonicTime time_;
+    // Random tie-breaker for alarms scheduled for the same monotonic time used to mimic
+    // non-deterministic execution of real alarms scheduled for the same wall time.
+    uint64_t randomness_;
+    Alarm* alarm_;
+
+    friend bool operator<(const AlarmRegistration& lhs, const AlarmRegistration& rhs) {
+      if (lhs.time_ != rhs.time_) {
+        return lhs.time_ < rhs.time_;
```

Comment:
if lhs.time_ > rhs.time_ shouldn't we return false here? We don't want to then fall thru to `.randomness` then right?

Reply:
We only fallback to randomness when the two times are equal.  The code already behaves as you suggest.

Reply:
indeed; my bad; this logic is fine.